processing/src/main/java/org/apache/carbondata/processing/newflow/DataLoadProcessorStep.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.carbondata.processing.newflow;
+
+import java.util.Iterator;
+
+import org.apache.carbondata.processing.newflow.exception.CarbonDataLoadingException;
+
+/**
+ * This base interface for data loading. It can do transformation jobs as per the implementation.
+ *
+ */
+public interface DataLoadProcessorStep {
+
+  /**
+   * The output meta for this step. The data returns from this step is as per this meta.
+   * @return
+   */
+  DataField[] getOutput();
+
+  /**
+   * Intialization process for this step.
+   * @param configuration
+   * @param child
+   * @throws CarbonDataLoadingException
+   */
+  void intialize(CarbonDataLoadConfiguration configuration, DataLoadProcessorStep child) throws
+      CarbonDataLoadingException;
+
+  /**
+   * Tranform the data as per the implemetation.
+   * @return Iterator of data
+   * @throws CarbonDataLoadingException
+   */
+  Iterator<Object[]> execute() throws CarbonDataLoadingException;
```

Comment:
For suppose if we are loading 50GB of csv files and each HDFS block size is 256MB then total number of partitions are 200. If we allow one task per partition then it would be 200 tasks. In carbondata one btree is created for each task. So if we allow all 200 tasks then it would be massively 200 btrees and it is not effective both in performance and memory wise. 
That is the reason why we pool multiple blocks per task in the current kettle implementation. And these blocks are processed parallely. We can take the same way and use iterator for each thread and returns array of iterator.

What do you mean by datanode-scope sorting? how to synchronize between multiple tasks?